lib/icingadb/redisconnection.cpp

```diff
@@ -338,21 +338,15 @@ void RedisConnection::WriteLoop(asio::yield_context& yc)
 	for (;;) {
 		m_QueuedWrites.Wait(yc);
 
-	WriteFirstOfHighestPrio:
-		for (auto& queue : m_Queues.Writes) {
-			if (m_SuppressedQueryKinds.find(queue.first) != m_SuppressedQueryKinds.end() || queue.second.empty()) {
-				continue;
-			}
-
-			auto next (std::move(queue.second.front()));
-			queue.second.pop();
+		while (!m_Queues.Writes.empty() && m_SuppressedQueryKinds.find(m_Queues.Writes.top().Priority) == m_SuppressedQueryKinds.end()) {
+			auto next (m_Queues.Writes.top());
+			m_Queues.Writes.pop();
 
 			WriteItem(yc, std::move(next));
-
-			goto WriteFirstOfHighestPrio;
 		}
 
 		m_QueuedWrites.Clear();
+		m_Queues.NextWriteOrder = 0;
```

Comment:
I said [_if the queue is empty_](https://github.com/Icinga/icinga2/pull/8557#pullrequestreview-567434481), not _unconditionally_. Explain with your own words why not checking whether empty might be a problem.

Reply:
Yes, because it can go out from the while loop even if ``m_QueueWrites`` is not empty, if one of the priorities in here are ``m_SuppressedQueryKinds``.